skills_ml/datasets/job_postings.py

```diff
@@ -33,15 +33,32 @@ def job_postings(s3_conn, quarter, s3_path):
     bucket = s3_conn.get_bucket(bucket_name)
     keys = bucket.list(prefix='{}/{}'.format(prefix, quarter))
     for key in keys:
-        logging.info('Extracting job postings from key {}'.format(key.name))
-        with tempfile.NamedTemporaryFile() as outfile:
-            retrier.call(key.get_contents_to_file, outfile, cb=log_download_progress)
-            outfile.seek(0)
-            for line in outfile:
-                yield line.decode('utf-8')
+        if isinstance(source, str):
+            if source.upper() in key.name or source.lower() == "all":
```

Comment:
This could get some false positives given the random filenames, right?

The source should always be at the beginning of the key name, or at the very least the beginning of the 'filename' part (not including the prefixes). You should be able to regex this out, or do `.startswith(source.upper())` on the correct part of the key.name.

Reply:
After looking at this some more: Can you just add this to the prefix on line 34?
Like:
`keys = bucket.list(prefix='{}/{}/{}_'.format(prefix, quarter, source.upper()))`